src/adapters/postproc.rs

```diff
@@ -145,30 +143,35 @@ pub fn postproc_prefix(line_prefix: &str, inp: impl AsyncRead + Send) -> impl As
     StreamReader::new(oup_stream)
 }
 
-/**
- * adds the prefix `Page N:` to each line,
- * where N starts at one and is incremented for each ASCII Form Feed character in the input stream.
- * (That's the format output by pdftotext)
- */
+/// Adds the prefix "Page N:" to each line,
+/// where N starts at one and is incremented for each ASCII Form Feed character in the input stream.
+/// ASCII form feeds are the page delimiters output by `pdftotext`.
 pub fn postproc_pagebreaks(line_prefix: &str, inp: impl AsyncRead) -> impl AsyncRead {
-    let line_prefix = line_prefix.to_string(); // clone since
-    let mut page_count = 1;
+    let form_feed = b'\x0c';
+    let regex = regex::bytes::Regex::new("\n").unwrap();
+    let mut page_count = 0;
+    let mut line_prefix = format!("\n{}Page {}:", line_prefix, page_count + 1);
 
-    panic!("todo!");
-    tokio::io::empty()
-    /*ByteReplacer {
-        inner: inp,
-        next_read: format!("{}Page {}:", line_prefix, page_count).into_bytes(),
-        haystacker: Box::new(|buf| memchr::memchr2(b'\n', b'\x0c', buf)),
-        replacer: Box::new(move |b| match b {
-            b'\n' => format!("\n{}Page {}:", line_prefix, page_count).into_bytes(),
-            b'\x0c' => {
-                page_count += 1;
-                format!("\n{}Page {}:", line_prefix, page_count).into_bytes()
+    let inp_stream = ReaderStream::new(inp);
+    let oup_stream = stream! {
+        yield Ok(Bytes::copy_from_slice(line_prefix.as_bytes()));
+        for await chunk in inp_stream {
+            match chunk {
+                Err(e) => yield Err(e),
+                Ok(chunk) => {
+                    let chunk_iter = chunk.split(|byte| byte == &form_feed);
+                    for sub_chunk in chunk_iter {
+                        if sub_chunk.contains(&b'\n') {
+                            yield Ok(Bytes::copy_from_slice(&regex.replace_all(&sub_chunk, line_prefix.as_bytes())));
+                            page_count += 1;
```

Comment:
i think you have a logic error here. the page count should increase for each chunk here not just if the chunk contains \n.
or rather, it should increase on each chunk except the first one i guess

Reply:
Right you are. I tried again in f19e173585c454355b9221059fed4dc1a9115792, where `page_count` is incremented when and only when a form feed character is encountered.

Reply:
uh i think that's still not right :D 

The stream would look like this (`\f` is form feed)

`Hello\nWorld\fFoo Bar\n\fTest`

splitting would result in:

`Hello\nWorld`,   `Foo Bar\n`, `Test`

The result should be

Page 1: Hello\nWorld
Page 2: Foo Bar\n
Page 3: Test

You don't just want to increase the page if the chunk is *just* form_feed (I think)

Reply:
(updated, my example was wrong